pkg/build/api/v1/types.go

```diff
@@ -171,6 +176,31 @@ type BuildSource struct {
 	SourceSecret *kapi.LocalObjectReference `json:"sourceSecret,omitempty" description:"supported auth methods are: ssh-privatekey"`
 }
 
+// ImageSource describes an image that is used as source for the build
+type ImageSource struct {
+	// From is a reference to an ImageStream tag or a DockerImage to
+	// copy source from
+	From kapi.ObjectReference `json:"from" description:"image reference"`
+
+	// Paths is a list of source and destination paths to copy from the image
+	Paths []ImageSourcePath `json:"paths" description:"paths to copy from image"`
```

Comment:
what happens when one path does not exists in the image, but others does? will that fail the build or the path will be skipped?